### PR TITLE
docs: reporter guide

### LINKED ISSUE #296: Docs: Reporter Guide
### Clear and concise description of the problem

See https://github.com/vitest-dev/vitest/blob/main/packages/vitest/src/types/reporter.ts

### Suggested solution

A new guide

### Alternative

_No response_

### Additional context

_No response_

### Validations

- [X] Follow our [Code of Conduct](https://github.com/vitest-dev/vitest/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guidelines](https://github.com/vitest-dev/vitest/blob/main/CONTRIBUTING.md).
- [X] Read the [docs](https://vitest.dev/guide).
- [X] Check that there isn't already an issue that request the same feature to avoid creating a duplicate.

### PR DESCRIPTION
**Pull request by allisons11**
### Description

Resolves #296 - guide for basic usage of reporters. Includes:
- Usage via CLI and config file
- Output files
- Combining reporters
- Summary of built-in reporters (default, basic, verbose, dot, json, junit, html, tap, tap-flat, hanging-process) with examples of output for test results
- Custom reporters - I haven't gone into detail about the Reporter API beyond linking to the instance

### Please don't delete this checklist! Before submitting the PR, please make sure you do the following:
- [x] It's really useful if your PR references an issue where it is discussed ahead of time. If the feature is substantial or introduces breaking changes without a discussion, PR might be closed.
- [ ] Ideally, include a test that fails without this PR but passes with it.
- [x] Please, don't make changes to `pnpm-lock.yaml` unless you introduce a new test example.

### Tests
- [x] Run the tests with `pnpm test:ci`.

### Documentation
- [ ] If you introduce new functionality, document it. You can run documentation with `pnpm run docs` command.

### Changesets
- [x] Changes in changelog are generated from PR name. Please, make sure that it explains your changes in an understandable manner. Please, prefix changeset messages with `feat:`, `fix:`, `perf:`, `docs:`, or `chore:`.

### CODE REVIEW

File: docs/guide/reporters.md [hunk 1]
```diff
@@ -0,0 +1,240 @@
...
+```
+
+```ts
+export default defineConfig({
+  test: {
+    reporters: ['json', 'default'],
+    outputFile: './test-output.json'
+  }
+});
+```
```
[the remark this answers is not in the document]
Comment (allisons11) [reply docs/guide/reporters.md:62]: Good call - will add!

File: docs/guide/reporters.md [hunk 2]
```diff
@@ -0,0 +1,427 @@
...
+export default defineConfig({
+  test: {
+    reporters: ['verbose']
+  },
+});
+```
+
+## Reporter Output
+
+By default, Vitest's reporters will print their output to the terminal. When using the `json`, `html`, or `junit` reporters, you can instead write your tests' output to a file by including an `outputFile` [configuration option](https://vitest.dev/config/#outputfile) either in your Vite configuration file or via CLI.
```
Comment (AriPerkkio) [inline docs/guide/reporters.md:30]:
```suggestion
By default, Vitest's reporters will print their output to the terminal. When using the `json`, `html` or `junit` reporters, you can instead write your tests' output to a file by including an `outputFile` [configuration option](https://vitest.dev/config/#outputfile) either in your Vite configuration file or via CLI.
```

**Review by AriPerkkio — CHANGES_REQUESTED** [review]
Overall the contents look good to me. 👍 

This also needs to be defined in `docs/.vitepress/config.ts` so that it appears in the left side menu. Between `'Test Filtering'` and `'Coverage'` would be good spot I think. 

https://github.com/vitest-dev/vitest/blob/4293e1b15111316b8ef622d04f4e5d8b70ad9474/docs/.vitepress/config.ts#L140-L143

**Review by AriPerkkio — APPROVED** [review]

File: docs/guide/reporters.md [hunk 3]
```diff
@@ -0,0 +1,428 @@
...
+  test: {
+    reporters: ['json'],
+    outputFile: './test-output.json'
+  },
+})
+```
+
+## Combining Reporters
+
+You can use multiple reporters simultaneously to print your tests results in different formats. For example:
```
Comment (dammy001) [inline docs/guide/reporters.md:49]:
```suggestion
You can use multiple reporters simultaneously to print your test results in different formats. For example:
```

File: docs/guide/reporters.md [hunk 4]
```diff
@@ -0,0 +1,428 @@
...
+```ts
+export default defineConfig({
+  test: {
+    reporters: ['json', 'default'],
+    outputFile: './test-output.json'
+  },
+})
+```
+
+The above example will both print the test results to the terminal in the default style, and write them as JSON to the designated output file.
```
Comment (dammy001) [inline docs/guide/reporters.md:64]:
```suggestion
The above example will both print the test results to the terminal in the default style and write them as JSON to the designated output file.
```

File: docs/guide/reporters.md [hunk 5]
```diff
@@ -0,0 +1,428 @@
...
+
+This example will write separate JSON and XML reports as well as printing a verbose report to the terminal.
+
+## Built-in Reporters
+
+### Default reporter
+
+By default (i.e. if no reporter is specified), Vitest will display results for each test suite hierarchically as they run, and then collapse after a suite passes. When all tests have finished running, the final terminal output will display a summary of results and details of any failed tests.
+
+```bash
```
Comment (dammy001) [inline docs/guide/reporters.md:86]: do we need an example for the default reporter? I don't think there's a need to pass --reporter=default. maybe the output example is enough

File: docs/guide/reporters.md [hunk 6]
```diff
@@ -0,0 +1,428 @@
...
+
+ Test Files  2 passed (2)
+      Tests  4 passed (4)
+   Start at  12:34:32
+   Duration  1.26s (transform 35ms, setup 1ms, collect 90ms, tests 1.47s, environment 0ms, prepare 267ms)
+```
+
+### Basic reporter
+
+The `basic` reporter displays the tests files that have run and a summary of results after the entire suite has finished running. Individual tests are not included in the report unless they fail.
```
Comment (dammy001) [inline docs/guide/reporters.md:121]:
```suggestion
The `basic` reporter displays the test files that have run and a summary of results after the entire suite has finished running. Individual tests are not included in the report unless they fail. 
```

File: docs/guide/reporters.md [hunk 7]
```diff
@@ -0,0 +1,428 @@
...
+      "message": "",
+      "name": "/root-directory/__tests__/test-file-1.test.ts"
+    }
+  ]
+}
+```
+
+### HTML reporter
+
+Generates an HTML file to view tests results through an interactive GUI. After the file has been generated, Vitest will keep a local development server running and provide a link to view the report in a browser.
```
Comment (dammy001) [inline docs/guide/reporters.md:304]:
```suggestion
Generates an HTML file to view test results through an interactive GUI. After the file has been generated, Vitest will keep a local development server running and provide a link to view the report in a browser.
```

File: docs/guide/reporters.md [hunk 8]
```diff
@@ -0,0 +1,428 @@
...
+export default defineConfig({
+  test: {
+    reporters: ['hanging-process']
+  },
+})
+```
+
+## Custom reporters
+
+You can use third-party custom reporters installed from NPM by specifying their package name in the reporters option:
```
Comment (dammy001) [inline docs/guide/reporters.md:409]:
```suggestion
You can use third-party custom reporters installed from NPM by specifying their package name in the reporters' option:
```

**Review by dammy001 — COMMENTED** [review]
Great work!

File: docs/guide/reporters.md (hunk 5 above)
Comment (allisons11) [reply docs/guide/reporters.md]: Agreed, I don't think it's necessary. Deleted the terminal and config examples

**Comment by sheremet-va** [comment]
Thank you very much for this detailed guide 🙏🏻

**Pull request merged**